From ee120f067124f6aaad0271328c4c8182ad332ece Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Thu, 11 Oct 2018 12:38:03 +0300
Subject: [PATCH 76/86] compulab-imx8mq: General refactoring

1) Add common SOM interfaces and its pinmuxes.
2) Remove SB-iMX8 staff.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 arch/arm64/boot/dts/compulab/compulab-imx8mq.dtsi | 29 +++++++----------------
 1 file changed, 8 insertions(+), 21 deletions(-)

diff --git a/arch/arm64/boot/dts/compulab/compulab-imx8mq.dtsi b/arch/arm64/boot/dts/compulab/compulab-imx8mq.dtsi
index 02a0af9..e402e16 100644
--- a/arch/arm64/boot/dts/compulab/compulab-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/compulab/compulab-imx8mq.dtsi
@@ -185,15 +185,15 @@
 
 		pinctrl_i2c1: i2c1grp {
 			fsl,pins = <
-				MX8MQ_IOMUXC_I2C1_SCL_I2C1_SCL			0x4000007f
-				MX8MQ_IOMUXC_I2C1_SDA_I2C1_SDA			0x4000007f
+				MX8MQ_IOMUXC_I2C1_SCL_I2C1_SCL	0x4000007f
+				MX8MQ_IOMUXC_I2C1_SDA_I2C1_SDA	0x4000007f
 			>;
 		};
 
 		pinctrl_i2c2: i2c2grp {
 			fsl,pins = <
-				MX8MQ_IOMUXC_I2C2_SCL_I2C2_SCL			0x4000007f
-				MX8MQ_IOMUXC_I2C2_SDA_I2C2_SDA			0x4000007f
+				MX8MQ_IOMUXC_I2C2_SCL_I2C2_SCL	0x4000007f
+				MX8MQ_IOMUXC_I2C2_SDA_I2C2_SDA	0x4000007f
 			>;
 		};
 
@@ -204,10 +204,10 @@
 			>;
 		};
 
-		touchscreen_pins: touchscreen-pins {
+		pinctrl_i2c4: i2c4grp {
 			fsl,pins = <
-				MX8MQ_IOMUXC_NAND_CE3_B_GPIO3_IO4 0x17059 /* P21.34 IRQ */
-				MX8MQ_IOMUXC_NAND_CE2_B_GPIO3_IO3 0x17059 /* P21.32 RST */
+				MX8MQ_IOMUXC_I2C4_SCL_I2C4_SCL	0x4000007f
+				MX8MQ_IOMUXC_I2C4_SDA_I2C4_SDA	0x4000007f
 			>;
 		};
 
@@ -523,20 +523,7 @@
     clock-frequency = <400000>;
     pinctrl-names = "default";
     pinctrl-0 = <&pinctrl_i2c3>;
-    status = "okay";
-    goodix_ts@5d {
-        compatible = "goodix,gt911";
-        reg = <0x5d>;
-
-        interrupt-parent = <&gpio3>;
-        interrupts = <3 IRQ_TYPE_EDGE_FALLING>;
-
-        pinctrl-names = "default";
-        pinctrl-0 = <&touchscreen_pins>;
-
-        irq-gpios = <&gpio3 3 GPIO_ACTIVE_HIGH>;
-        reset-gpios = <&gpio3 4 GPIO_ACTIVE_HIGH>;
-    };
+    status = "disabled";
 };
 
 &pwm2 {
-- 
1.9.1

